Framework/Algorithms/test/CreateDetectorTableTest.h

```diff
@@ -98,6 +100,11 @@ class CreateDetectorTableTest : public CxxTest::TestSuite {
     TS_ASSERT_THROWS_NOTHING(alg.execute();)
     TS_ASSERT(alg.isExecuted());
 
+    // Check that a missing efixed value throws
+    const auto &spectrumInfo = inputWS->spectrumInfo();
+    std::shared_ptr<const IDetector> detector(&spectrumInfo.detector(0), Mantid::NoDeleting());
+    TS_ASSERT_THROWS_ANYTHING(inputWS->getEFixed(detector));
```

Comment:
I don't think this is testing your bug fix, it would still pass without the code changes.